### PR TITLE
Convenience method for action return schemas

### PR DESCRIPTION
**Pull request by patrick-codaio**

### CODE REVIEW

File: schema.ts [hunk 1]
```diff
@@ -1256,3 +1256,15 @@ export function makeReferenceSchemaFromObjectSchema(
...
+
+/**
+ * Convenience for defining the result schema for an action. The identity enables Coda to
+ * update the corresponding sync table row, if it exists.
+ * You could add the identity directly, but that would make the schema less re-usable.
+ */
+export function withIdentity(schema: GenericObjectSchema, identityName: string): GenericObjectSchema {
+  return makeObjectSchema({
+    ...schema,
+    identity: {name: ensureExists(identityName)},
```
Comment (jonathan-codaio) [inline schema.ts:1268]: Do you want ensureNonEmptyString() here? As-is I think this is a no-op.

File: schema.ts [hunk 2]
```diff
@@ -1256,3 +1256,15 @@ export function makeReferenceSchemaFromObjectSchema(
...
 }
+
+/**
+ * Convenience for defining the result schema for an action. The identity enables Coda to
+ * update the corresponding sync table row, if it exists.
+ * You could add the identity directly, but that would make the schema less re-usable.
+ */
+export function withIdentity(schema: GenericObjectSchema, identityName: string): GenericObjectSchema {
+  return makeObjectSchema({
+    ...schema,
```
Comment (jonathan-codaio) [inline schema.ts:1267]: Do we want to deep copy the schema, in case the maker mutates it?

Comment (patrick-codaio) [reply schema.ts]: Oh, duh. Thanks! Done.

File: schema.ts (hunk 1 above)
Comment (patrick-codaio) [reply schema.ts]: Yup, done!

**Review by jonathan-codaio — APPROVED** [review]
Would consider adding a CHANGELOG entry too.

**Pull request merged**